lib/jeckle/rest_actions.rb

```diff
@@ -13,11 +13,11 @@ def find(id)
       end
 
       def search(params = {})
-        response   = run_request(resource_name, params).response.body || []
+        response   = run_request(params.delete[:resource_name] || resource_name, params).response.body || []
```

Comment:
Sorry for being late, guys.

First of all, I believe this `XXX.json` suffix should NOT be part of the `resource_name`.

If you think about it, it's an exception rule for some of the listing URLs, like `GET /photos.json` or `GET /products/categories.json`. If by any chance a POST action is created for the same resource, the URLs are gonna be weird/wrong: `POST /photos.json/create` and `POST /products/categories.json/create`.

Taking that into account, the solution that @brennovich suggested and @cairesr implemented sounds like a good alternative, it's more isolated.

But in a near future I think this case should be handled by the custom action DSL. Something like this:

```ruby
class ProductCategory
  include Jeckle::Resource

  api :whatever

  action :search, method: :get, on: :collection, path: '', suffix: '.json'
end

ProductCategory.search
>> /product_categories.json
```

Please tell me what you think @brennovich @tomas-stefano.